persistence/sql/persister_oauth2.go

```diff
@@ -231,23 +231,23 @@ func (p *Persister) findSessionBySignature(ctx context.Context, rawSignature str
 			return errorsx.WithStack(fosite.ErrNotFound)
 		} else if err != nil {
 			return sqlcon.HandleError(err)
-		} else if !r.Active && table == sqlTableCode {
+		} else if !r.Active {
 			fr, err = r.toRequest(ctx, session, p)
 			if err != nil {
 				return err
-			} else {
+			} else if table == sqlTableCode {
 				return errorsx.WithStack(fosite.ErrInvalidatedAuthorizeCode)
+			} else if table == sqlTableRefresh {
+				return errorsx.WithStack(fosite.ErrInactiveToken)
 			}
-		} else if !r.Active {
-			return errorsx.WithStack(fosite.ErrInactiveToken)
```

Comment:
This introduces a serious issue as it is not covering the case where `r.Active` is false and table is none of the options. Please add a test for this!

Reply:
What would be preferred behaviour if something is inactive? Here it falls through as before.

Reply:
Before, if `r.Active` was false, you would get `fosite.ErrInactiveToken`. Now, you won't get that and end up at `r.toRequest`. You need to add a `return errorsx.WithStack(fosite.ErrInactiveToken)` after

```
 			} else if table == sqlTableRefresh {
 				return errorsx.WithStack(fosite.ErrInactiveToken)
 			}
```

Reply:
Sorry I overlooked that. I don't have to modify this method at all, it was working as required.